### PR TITLE
Expose a parameter on the DynamoDB test hooks for running the DynamoDB instance in-memory

### PR DESCRIPTION
**Pull request by spiffytech**

### CODE REVIEW

File: src/dynamodb.js [hunk 1]
```diff
@@ -66,14 +66,11 @@ async function destroyTables (dynamoClient, uniqueIdentifier) {
...
-  }
-
-  const docker = new Docker();
-  const environment = new Environment();
-
+/**
+ * @param params.inMemory Whether to run local DynamoDB in in-memory mode.
+ * 'false' persists data to disk.
+ */
+async function launchDynamoContainer ({ inMemory } = { inMemory: false }, docker = new Docker()) {
```
Comment (mdlavin) [inline src/dynamodb.js]: Nit: can the docker option be moved into the first argument hash?

File: src/dynamodb.js [hunk 2]
```diff
@@ -126,11 +139,17 @@ exports.tableSchema = (schema) => {
...
 };
 
-function dynamoDBTestHooks (useUniqueTables = false) {
+/**
+ * @param {boolean} useUniqueTables
+ * @param {object} params
+ * @param params.inMemory Determines whether to run the local DynamoDB instance
+ * in in-memory mode, or to persist the data to disk
+ */
+function dynamoDBTestHooks (useUniqueTables = false, { inMemory } = { inMemory: false }) {
```
Comment (mdlavin) [inline src/dynamodb.js]: Do the TypeScript type files need to be updated to include this new argument?

Comment (spiffytech) [reply src/dynamodb.js]: Done.

File: src/dynamodb.js (hunk 1 above)
Comment (spiffytech) [reply src/dynamodb.js]: Done.

**Comment by mdlavin** [comment]
Looks like there are uncovered lines of code with the latest changes

**Comment by mdlavin** [comment]
Can you add updates to the typescript .d.ts files for the new API changes?

**Comment by spiffytech** [comment]
I got the unit test coverage back up to 100%.

I updated `index.d.ts` with the `inMemory` parameter. I left out the `docker` parameter since that should only be used internally in tests.

File: src/dynamodb.js [hunk 3]
```diff
@@ -66,14 +66,13 @@ async function destroyTables (dynamoClient, uniqueIdentifier) {
...
-  const docker = new Docker();
-  const environment = new Environment();
-
+/**
+ * @param {object} opts
+ * @param {object} opts.docker Used to mock the Docker library in tests
+ * @param {boolean} opts.inMemory Whether to run local DynamoDB in in-memory mode.
+ * 'false' persists data to disk.
+ */
+async function launchDynamoContainer ({ docker = new Docker(), inMemory = false } = { docker: new Docker(), inMemory: false }) {
```
Comment (mdlavin) [inline src/dynamodb.js]: Question, would `{ docker = new Docker(), inMemory = false } = { }` achieve the same thing without duplication?

File: src/index.d.ts [hunk 4]
```diff
@@ -38,8 +38,8 @@ declare namespace dynamodb {
   export function tableSchema(
     schema: ReadonlyArray<aws.DynamoDB.Types.CreateTableInput>
   ): void;
-  export function dynamoDBTestHooks(useUniqueTables?: boolean): Hooks;
-  export function useDynamoDB(test: TestInterface, useUniqueTables?: boolean): void;
+  export function dynamoDBTestHooks(useUniqueTables?: boolean, {inMemory}?: {inMemory: boolean}): Hooks;
```
Comment (mdlavin) [inline src/index.d.ts]: Does the `{inMemory}?: {inMemory: boolean}` syntax work? I haven't seen that before. I expected to see `args?: {inMemory: boolean}`

**Pull request merged**